### DISCUSSION TITLE
Certificate using with ldaps

### DISCUSSION THREAD

**Discussion by guncer** [dsbenghe/Novell.Directory.Ldap.NETStandard#183] [General]
Hello,
we used ldaps connection the production servers. It works when the certificate is added under trusted root certificate.  but we can not understand one thing. 
we reviewed the source code , but we could not find any certificate name or path  set section in the code. How does it know which root certificate is used when connecting the ldap?  In the code, we see below line, however, we do not see any certificates added to this list. what are we missing? I would appreciate your help
`
        private List<X509Certificate> _clientCertificates = new List<X509Certificate>();`
  

Best Regards,

**Comment by se** [comment]
Hi, @guncer.

Certificates comes from your Operating System's Certificate Store. When a connection needs a trust / validate for a SSL / TLS connection, your target (which is your DC in this case) will send a certificate to your cleint. That certificate includes some information about like "Intermediate certificate authority" and "Root certificate authority". If your certification store doesn't include these authorities, your connection won't be accepted by your client (if you didn't modify anything about to certification validation).

It would be better to ask your direct achivement in this? Like, what do you want to do and for what?
